MapboxNavigation/CarPlayManager.swift

```diff
@@ -570,64 +573,74 @@ extension CarPlayManager: CPMapTemplateDelegate {
         } else {
             return
         }
-        
+
         let decelerationRate: CGFloat = 0.9
         let offset = CGPoint(x: velocity.x * decelerationRate / 4, y: velocity.y * decelerationRate / 4)
-        
+
         if let toCamera = cameraShouldPan(to: offset, mapView: mapView) {
             mapView.setCamera(toCamera, animated: true)
         }
     }
-    
+
     func cameraShouldPan(to endPoint: CGPoint, mapView: NavigationMapView) -> MGLMapCamera? {
         let mapView = mapView
         let camera = mapView.camera
         let centerPoint = CGPoint(x: mapView.bounds.midX, y: mapView.bounds.midY)
         let endCameraPoint = CGPoint(x: centerPoint.x - endPoint.x, y: centerPoint.y - endPoint.y)
-        
+
         camera.centerCoordinate = mapView.convert(endCameraPoint, toCoordinateFrom: mapView)
-        
+
         return camera
     }
-    
+
     public func mapTemplate(_ mapTemplate: CPMapTemplate, didEndPanGestureWithVelocity velocity: CGPoint) {
         if let navigationViewController = currentNavigator, mapTemplate == navigationViewController.mapTemplate {
             return
         }
         mapTemplate.mapButtons.forEach { $0.isHidden = false }
     }
-    
+
     public func mapTemplateDidShowPanningInterface(_ mapTemplate: CPMapTemplate) {
         guard let carPlayMapViewController = self.carWindow?.rootViewController as? CarPlayMapViewController else {
             return
         }
         carPlayMapViewController.mapView.userTrackingMode = .none
     }
-    
+
     public func mapTemplate(_ mapTemplate: CPMapTemplate, panWith direction: CPMapTemplate.PanDirection) {
         guard let carPlayMapViewController = self.carWindow?.rootViewController as? CarPlayMapViewController else {
             return
         }
-        
+
         let mapView = carPlayMapViewController.mapView
         let camera = mapView.camera
-        
+
         mapView.userTrackingMode = .none
 
         var facing: CLLocationDirection = 0.0
-        
+
         if direction.contains(.right) {
             facing = 90
         } else if direction.contains(.down) {
             facing = 180
         } else if direction.contains(.left) {
             facing = 270
         }
-        
+
         let newCenter = camera.centerCoordinate.coordinate(at: CarPlayMapViewPanningIncrement, facing: facing)
         camera.centerCoordinate = newCenter
         mapView.setCamera(camera, animated: true)
     }
+
+    private func createRouteController(with route: Route) -> RouteController {
+        if self.simulatesLocations {
+            let locationManager = SimulatedLocationManager(route: route)
+            locationManager.speedMultiplier = self.simulatedSpeedMultipler
```

Comment:
Idle thought: what if we make 0 mean “don’t simulate”?

Reply:
For now I'd like to keep things as-is. But it's an interesting idea. I wonder if it would simplify or complicate setting this value from a UI. It might make sense with a slider or dial, but not with a button as we currently have (of course the multiplier isn't even exposed directly to CarPlay UI right now)